etrago/tools/plot.py

```diff
@@ -1153,6 +1306,119 @@ def nodal_production_balance(network, timesteps, scaling=0.00001):
 
     """
 
+    def import_gen_from_links(network):
+        from pypsa.networkclustering import aggregategenerators
+        from etrago.cluster.spatial import strategies_generators
```

Comment:
Imports should be always moved to the top of files

Reply:
Solved in db21a49